fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -30,6 +30,13 @@ import org.mozilla.fenix.wallpapers.Wallpaper
  * [Action] implementation related to [AppStore].
  */
 sealed class AppAction : Action {
+    /**
+     * [AppAction] dispatched to indicate that the store is initialized and
+     * ready to use. This action is dispatched automatically before any other
+     * action is processed. Its main purpose is to trigger initialization logic
+     * in middlewares. The action itself should have no effect on the [AppState].
+     */
+    object Init : AppAction()
```

Comment:
Looking at `BrowserStore`, the `InitAction` seems to be more of an object lifecycle than Android lifecycle thing that indicates that the middlewares should do anything they need to fetch initial state.

Personally, I think it adds a little more clarity to see the `InitAction` dispatched inline from the Store constructor rather than as a response to `FenixApplication`. Plus, I'm guessing we get stronger guarantees about action ordering if we dispatch before the constructor completes than we would otherwise.

I'm still not opposed, but if we make changes I think we should keep the two Stores' workflows in sync.